### PR TITLE
movie deletion

### LINKED ISSUE #195: Add Delete endpoint for movies
DELETE `/api/movies/{movie.id}`

### PR DESCRIPTION
**Pull request by frodopwns**
closes #195 

**What this PR does / why we need it**:
This PR will let users send delete requests for specific movies

- add controller for DELETE /api/movies/movie_id
- add deleteDocument to db interface
- adds deleteDocument to cosmos db client

**Special notes for your reviewer**:
TODO:
**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains tests

### CODE REVIEW

**Comment by frodopwns** [comment]
`/key` in our data needs to be converted to a string

**Comment by frodopwns** [comment]
This PR won't pass until the getById endpoint returns 404 for non existing movies

File: src/db/idatabaseprovider.ts [hunk 1]
```diff
@@ -21,6 +21,18 @@ export interface IDatabaseProvider {
                    collection: string,
                    content: any): Promise<RetrievedDocument>;
 
+    /**
+     * Runs the given query against CosmosDB.
+     * @param database The database the document is in.
+     * @param collection The collection the document is in.
+     * @param query The query to select the documents.
+     */
```
Comment (ericpera-msft) [inline src/db/idatabaseprovider.ts]: Can you change this comment block to properly reflect this function definition?

File: src/db/cosmosdbprovider.ts [hunk 2]
```diff
@@ -105,6 +115,35 @@ export class CosmosDBProvider {
         });
     }
 
+    /**
+     * Delete the given document.
+     * @param database The database the document is in.
+     * @param collection The collection the document is in.
+     * @param query The query to select the documents.
+     */
```
Comment (ericpera-msft) [inline src/db/cosmosdbprovider.ts]: Can you change this comment block to properly reflect this function implementation (just the params)?

File: src/db/idatabaseprovider.ts (hunk 1 above)
Comment (frodopwns) [reply src/db/idatabaseprovider.ts]: yes

File: src/db/cosmosdbprovider.ts (hunk 2 above)
Comment (frodopwns) [reply src/db/cosmosdbprovider.ts]: yes

**Pull request merged**